worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageMappingCallable.java

```diff
@@ -135,32 +132,32 @@ public CoverageMapping call() {
 
     }
 
-    private Set<Identifiers> extractAndFilter(org.hl7.fhir.dstu3.model.Bundle bundle) {
+    private Set<Identifiers> extractAndFilter(IBaseBundle bundle) {
         return getPatientStream(bundle)
                 .filter(patient -> skipBillablePeriodCheck || filterByYear(patient))
                 .map(this::extractPatientId)
                 .filter(Objects::nonNull)
                 .collect(toSet());
     }
 
-    private Stream<org.hl7.fhir.dstu3.model.Patient> getPatientStream(org.hl7.fhir.dstu3.model.Bundle bundle) {
-        return bundle.getEntry().stream().map(org.hl7.fhir.dstu3.model.Bundle.BundleEntryComponent::getResource)
-                .filter(resource -> resource.getResourceType() == org.hl7.fhir.dstu3.model.ResourceType.Patient)
-                .map(resource -> (org.hl7.fhir.dstu3.model.Patient) resource);
+    private Stream<IDomainResource> getPatientStream(IBaseBundle bundle) {
+        try {
+            return BundleUtils.getPatientStream(bundle, bfdClient.getVersion());
+        } catch (Exception ex) {
+            log.error("Unable to get patient stream", ex);
+            return null;
+        }
     }
```

Comment:
Not really used anywhere else so perhaps move to extractAndFilter method instead?

Reply:
Ok